sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -440,6 +441,47 @@ public BoundedWindow window() {
   }
 
   /**
+    * Annotation for declaring and dereferencing timers.
+    *
+    * <p><i>Not currently supported by any runner</i>.
+    *
+    * <p>To declare a timer, create a field of type {@link TimerSpec} annotated with a {@link
+    * TimerId}. To use the cell during processing, add a parameter of the appropriate {@link Timer}
+    * subclass to your {@link ProcessElement @ProcessElement} method, and annotate it with {@link
+    * TimerId}. See the following code for an example:
+    *
+    * <pre>{@code
+    * new DoFn<KV<Key, Foo>, Baz>() {
+    *   @TimerId("my-timer-id")
+    *   private final TimerSpec myTimer = TimerSpecs.timerForDomain(TimeDomain.EVENT_TIME);
+    *
+    *   @ProcessElement
+    *   public void processElement(
+    *       ProcessContext c,
+    *       @TimerId("my-timer-id") Timer myTimer) {
+    *     myTimer.setTimer(new Instant(...));
```

Comment:
+1 to @bjchambers' suggestion - relative timers make sense, like SplittableDoFn's ProcessContinuation.withResumeDelay().